frame/bounties/src/lib.rs

```diff
@@ -385,10 +475,7 @@ decl_module! {
 		/// - O(1).
 		/// # </weight>
 		#[weight = <T as Config>::WeightInfo::unassign_curator()]
-		fn unassign_curator(
-			origin,
-			#[compact] bounty_id: BountyIndex,
-		) {
+		fn unassign_curator(origin, #[compact] bounty_id: BountyIndex) {
```

Comment:
The case where a bounty gets unassigned while the bounty have some subbounties is not handled correctly

Reply:
ok I thought the subbounties should be unassigned too, but actually I understand it is not necessary, the next curator will inherit from the subbounties made by the previous one.

Then maybe `unassign_subcurrator` and `close_subbounty` should be able to be called even when the master bounty is not active. (though it shouldn't be callable by ex-master curator).

At some point I wonder what is better:
* allow master curator to split and award subbounty, and allow subcurator to claim award before master bounty is claimable:
  If master curator get unassigned, the achieved split works that are already claimed is not longer part of the bounty balance. That means if a new curator get on the tax we can't dismissed the subbounties he consider invalid.
* or make the award of the subbounties only claimable once the master bounty is claimable. In this case, if the curator get unassigned and accepted by another curator then the achieved subbounties are still pending and can be close by new curator.

I see we opted for the former strategy, so the SubbountyAwarded event is of same importance as BountyAwarded and if curator is doing bad split of the bounty balance, new curator cannot recover from it

Reply:
I think what is important in the current design is at least being able to call `close_subbounty` when the bounty is in state Funded or CuratorProposed.

Reply:
Thanks for this comment :+1: 

To be on the samepage, I use the same terminology used in the above description.

* Parent Bounty managed by Master Curator.
* Subbounty managed by Subcurator.

Indeed, after calling `fn unassign_curator()`, 

* Parent bounty enters into `Funded` state.
* All calls to subbounty returns error code of `UnexpectedStatus`.
* Except the function `fn claim_subbounty()`, only this function is callable at subbounty level.
 
Means subbounty which reached `PendingPayout` can claim the subbounty value after predefined  locking period.

Please correct me if I am wrong in my understanding, 

I hope your question is about, under emergency condition howto stop or close all bad active subbounties created by malicious Master curator ? 

Indeed, its valid concern. As suggested, is it OK to bypass parent bounty status check if the call originates from `RejectOrigin` ? Like below ...

![](https://i.imgur.com/kfC46Vp.png)

By this way its possible to close bad subbounties, except subbounties in `PendingPayout` state.

Reply:
Yes I think `RejectOrigin` should be able to call `close_subbounty` when parent bounty is `Active`, `CuratorProposed`, `Funded`. For the same reason we allow `RejectOrigin` to call `close_bounty`.
So it should do: unreserve the subcurator deposit and remove the subbounty.

Also thinking of it again: I think `RejectOrigin` should be able to call `unassign_subcurator` when parent bounty is `Active`, `CuratorProposed`, `Funded`. In case the subcurator is malicious or inactive. (i.e. for same reason it would call `unassign_curator`).

Once those are allowed I think the logic is good to me. What do you think ?

Reply:
Suggestion sounds good for me. Let us recheck with Bryan and take the call. Thanks for comments Guillaume  :+1:

Reply:
Agree.

Reply:
Thanks, Bryan,  as discussed will modify `fn close_subbounty()` &  `fn unassign_subcurator()`, 
such that, for the call originates from `RejectOrigin`, execution of the function is 
forced by ignoring the status check of parent bounty.

Reply:
Hello Bryan, Guillaume, Changes are available for review, Please share your comments.